### PR TITLE
Remaining mailets 1

### PR DESCRIPTION
**Pull request by mbaechler**

### CODE REVIEW

File: mailet/standard/src/main/java/org/apache/james/transport/mailets/Null.java [hunk 1]
```diff
@@ -26,25 +26,15 @@
 
 /**
  * Simplest Mailet which destroys any incoming messages.
- *
  */
 public class Null extends GenericMailet {
 
-    /**
-     * Set this mail to GHOST state, indicating that no further processing 
-     * should take place.
```
Comment (rouazana) [inline mailet/standard/src/main/java/org/apache/james/transport/mailets/Null.java]: maybe you should keep this comment somewhere?

File: mailet/standard/src/test/java/org/apache/james/transport/mailets/SetMailAttributeTest.java [hunk 2]
```diff
@@ -31,38 +30,76 @@
...
+        
+        mailet.service(mail);
 
-        assertEquals("true", mockedMail.getAttribute(ATTRIBUTE_NAME1));
-        assertEquals("true", mockedMail.getAttribute(ATTRIBUTE_NAME2));
+        assertThat(mail.getAttribute("org.apache.james.junit1")).isEqualTo("bar");
+    }
+    
+    @Test
+    public void shouldThrownWhenNullProperty() throws MessagingException {
```
Comment (rouazana) [inline mailet/standard/src/test/java/org/apache/james/transport/mailets/SetMailAttributeTest.java]: s/Thrown/Throw/

File: mailet/standard/src/test/java/org/apache/james/transport/mailets/SetMailAttributeTest.java [hunk 3]
```diff
@@ -31,38 +30,76 @@
...
+        FakeMailetConfig mailetConfig = new FakeMailetConfig("Test", FakeMailContext.defaultContext());
+        expectedException.expect(NullPointerException.class);
+        mailetConfig.setProperty(null, "bar");
+    }
+    
+    @Test
+    public void shouldThrownWhenNullValue() throws MessagingException {
```
Comment (rouazana) [inline mailet/standard/src/test/java/org/apache/james/transport/mailets/SetMailAttributeTest.java]: idem

File: mailet/standard/src/test/java/org/apache/james/transport/mailets/NullTest.java [hunk 4]
```diff
@@ -20,54 +20,31 @@
...
     }
 
-    // test if the right state was set
-    public void testNullMailet() throws MessagingException {
-        setupMockedMail(mockedMimeMessage);
-        setupMailet();
-
-        mailet.service(mockedMail);
+    @Test
+    public void shouldChangeStateToGuest() throws MessagingException {
```
Comment (aduprat) [inline mailet/standard/src/test/java/org/apache/james/transport/mailets/NullTest.java]: s/Guest/Ghost

File: mailet/standard/src/main/java/org/apache/james/transport/mailets/Null.java (hunk 1 above)
Comment (mbaechler) [inline mailet/standard/src/main/java/org/apache/james/transport/mailets/Null.java]: I though it says the same as "Simplest Mailet which destroys any incoming messages.", it's why I removed it. Do you still think it's useful ?

Comment (rouazana) [inline mailet/standard/src/main/java/org/apache/james/transport/mailets/Null.java]: you could add "by setting their state to GHOST" in the mailet description.

Comment (mbaechler) [inline mailet/standard/src/main/java/org/apache/james/transport/mailets/Null.java]: ok

File: mailet/standard/src/test/java/org/apache/james/transport/mailets/SetMailAttributeTest.java (hunk 2 above)
Comment (mbaechler) [inline mailet/standard/src/test/java/org/apache/james/transport/mailets/SetMailAttributeTest.java]: ok

File: mailet/standard/src/test/java/org/apache/james/transport/mailets/SetMailAttributeTest.java (hunk 3 above)
Comment (mbaechler) [inline mailet/standard/src/test/java/org/apache/james/transport/mailets/SetMailAttributeTest.java]: ok

File: mailet/standard/src/test/java/org/apache/james/transport/mailets/NullTest.java (hunk 4 above)
Comment (mbaechler) [inline mailet/standard/src/test/java/org/apache/james/transport/mailets/NullTest.java]: ok

**Pull request merged**